crates/storage/src/transactional.rs

```diff
@@ -233,10 +307,9 @@ where
     S: KeyValueInspect<Column = Column>,
 {
     fn get_from_changes(&self, key: &[u8], column: Column) -> Option<&WriteOperation> {
-        let k = key.to_vec();
         self.changes
             .get(&column.id())
-            .and_then(|btree| btree.get(&k))
+            .and_then(|btree| btree.get(key))
```

Comment:
We use property that collections work with `Q` generic. The `ReferenceBytesKey` implements `Borrow<[u8]>` that allows usage of `&[u8]` for lookups. It is very important for `InMemoryTransaction` since `get` is mostly used method=)


<img width="759" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/d3e967e1-9eef-40f6-b401-b16b32b19682">